vaccine_feed_ingest/stages/load.py

```diff
@@ -244,22 +248,32 @@ def _is_match(source: location.NormalizedLocation, candidate: dict) -> bool:
         else set()
     )
 
-    candidate = candidate["properties"]
+    candidate_props = candidate["properties"]
+
     candidate_links = (
-        set(candidate["concordances"]) if "concordances" in candidate else set()
+        set(candidate_props["concordances"])
+        if "concordances" in candidate_props
+        else set()
     )
     shared_links = source_links.intersection(candidate_links)
 
     if len(shared_links) > 0:
         # Shared concordances, mark as match
         return True
 
-    if candidate["full_address"] is not None and source.address is not None:
-        if canonicalize_address(
-            get_full_address(source.address)
-        ) == canonicalize_address(candidate["full_address"]):
-            # Canonicalized address matches, mark as match
-            return True
+    # Don't match locations with different providers
+    provider_matches = is_provider_similar(source, candidate, threshold=0.7)
+    if provider_matches is not None and provider_matches is False:
+        return False
+
+    # If there are phone numbers and the phone numbers don't match then fail to match
+    phone_matches = has_matching_phone_number(source, candidate)
+    if phone_matches is not None and phone_matches is False:
+        return False
```

Comment:
I wonder how common it is to see candidates which _should_ match but where we have a different phone number than the source.

That said, seems like a good rule at this point!

Reply:
I am following the Velma channel, and this was inspired by this case of exactly matching address, but different clinics: https://discord.com/channels/799147121357881364/839741667560259604/839748175204188180